lib/galaxy/model/store/__init__.py

```diff
@@ -2128,12 +2128,14 @@ def export_history(
         datasets = query.all()
         for dataset in datasets:
             dataset.annotation = get_item_annotation_str(sa_session, history.user, dataset)
-            add_dataset = (dataset.visible or include_hidden) and (not dataset.deleted or include_deleted)
+            should_include_file = (dataset.visible or include_hidden) and (not dataset.deleted or include_deleted)
             if dataset.id in self.collection_datasets:
```

Comment:
wouldn't this include datasets that are deleted?
should this be something like
```suggestion
            if not dataset.deleted and dataset.id in self.collection_datasets:
```

Reply:
Hmmm... then I guess the whole condition about collections is redundant as it will be covered by `(not dataset.deleted or include_deleted)` in the previous line.
Do you think there could be a reason to "force" include deleted dataset if they are part of a collection? If not, we can completely remove this condition.

Reply:
`add_dataset = (dataset.visible or include_hidden)` still makes sense, you only need to include the dataset if that is true or if the dataset is part of a collection

Reply:
> Do you think there could be a reason to "force" include deleted dataset if they are part of a collection? 

no, can't think of a reason to do that.